tensorflow/python/ops/math_ops.py

```diff
@@ -4857,7 +4857,177 @@ def sparse_segment_sum(
+  array([[[0, 1],
+          [1, 0]],
+         [[0, 0],
+          [1, 0]]], dtype=int32)>, <tf.Tensor: shape=(2, 2), dtype=float32, numpy=
+  array([[ 75.75, 173.  ],
+         [381.5 , 524.5 ]], dtype=float32)>, 
+  <tf.Tensor: shape=(2,), dtype=int32, numpy=array([2, 2], dtype=int32)>)
+
+  Args:
+    indices: `tf.Tensor` containing coordinates for the rows and columns to be multiplied.
+      Must have rank > 1.
+    values: `tf.Tensor` containing the values to be scaled and added to the sampled dot product.
+    dense_shape: `tf.Tensor` defining the dense shape of the output.
+    mat1: `tf.Tensor` to be multiplied. Must have rank > 1.
+    mat2: `tf.Tensor` to be multiplied. Must have rank > 1.
+    beta: Number to be multipled with `values`. Defaults to 1.0.
+    alpha: Number to be multiplied with the sampled dot product of `mat1` and `mat2`.
+      Defaults to 1.0.
+    output_type: the output datatype if needed. Defaults to float32.
+
+  Returns:
+    Three `tf.Tensor` objects making up the components of a `SparseTensor` representing the result
+    of the operation.
+
+    output = `alpha` * (`mat1` @ `mat2`) * spy(`indices`) + `beta` * `values`
+
+    The function `spy(indices)` is the sparsity pattern matrix derived from `indices`. 
+  """
+  if context.executing_eagerly():
+    if not isinstance(indices, ops.EagerTensor):
+      indices = ops.convert_to_tensor(indices)
+    if not isinstance(values, ops.EagerTensor):
+      values = ops.convert_to_tensor(values)
+    if not isinstance(mat1, ops.EagerTensor):
+      mat1 = ops.convert_to_tensor(mat1)
+    if not isinstance(mat2, ops.EagerTensor):
+      mat2 = ops.convert_to_tensor(mat2)
+  else:
+    if not isinstance(indices, tensor_lib.Tensor):
+      indices = ops.convert_to_tensor(indices)
+    if not isinstance(values, tensor_lib.Tensor):
+      values = ops.convert_to_tensor(values)
+    if not isinstance(mat1, tensor_lib.Tensor):
+      mat1 = ops.convert_to_tensor(mat1)
+    if not isinstance(mat2, tensor_lib.Tensor):
+      mat2 = ops.convert_to_tensor(mat2)
+
+  if values.dtype != output_type:
+    values = cast(values, output_type)
+  if mat1.dtype != output_type:
+    mat1 = cast(mat1, output_type)
+  if mat2.dtype != output_type:
+    mat2 = cast(mat2, output_type)
+
+  dense_rows = mat1.shape[-2]
+  dense_cols = mat2.shape[-1]
+
+  # TODO(mattbahr): use dense_shape to validate the shapes of mat1 and mat2
+  dense_shape = constant_op.constant([dense_rows, dense_cols])
+
+  # Extract row and column indices
+  batch_indices = indices[..., :-2]
+  row_indices = indices[..., :-1]
+  col_indices = array_ops.concat([batch_indices, 
+                                  indices[..., -1:]], axis=-1)
+
+  # Extract rows and columns 
+  rows = array_ops.gather_nd(mat1, row_indices, 
+                             batch_dims=mat1.ndim - 2)
+  cols = array_ops.gather_nd(array_ops.matrix_transpose(mat2), col_indices, 
+                             batch_dims=mat2.ndim - 2)
```

Comment:
A little unclear on what works/doesn't work in graph/eager modes. This seems to be ok with my test cases though